test/automation/src/sql/notebook.ts

```diff
@@ -95,4 +103,20 @@ export class NotebookToolbar {
 		const kernelDropdownValue = `${NotebookToolbar.toolbarSelector} select[id="kernel-dropdown"][title="${kernel}"]`;
 		await this.code.waitForElement(kernelDropdownValue, undefined, 3000); // wait up to 5 minutes for kernel change
 	}
+
+	async trustNotebook(): Promise<void> {
+		const trustedButton = `${NotebookToolbar.toolbarSelector} a[class="action-label codicon notebook-button masked-icon icon-shield"]`;
+		const notTrustedButton = `${NotebookToolbar.toolbarSelector} a[class="action-label codicon notebook-button masked-icon icon-shield-x"]`;
+
+		await this.code.waitAndClick(NotebookToolbar.toolbarSelector);
+		if (this.code.waitForElement(notTrustedButton)) {
```

Comment:
Probably ok for your scenario - but I would probably expect this function to just no-op if it's already trusted. Currently though it seems like it'll time out waiting for the not trusted button if that's the case though.

Reply:
I updated the method so that it gets all the buttons in the toolbar and checks if any of them have the trusted/not trusted icon class. If the trusted icon is present then no-op else click the not trusted icon.